chat-plugins/calculator.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+function isNumeric(str) {
+	return !isNaN(parseFloat(str));
+}
+
+function parseMathematicalExpression(infix) {
+	// Shunnting-yard Algorithm -- https://en.wikipedia.org/wiki/Shunting-yard_algorithm
+	let outputQueue = [];
+	let operatorStack = [];
+	let operators = {
+		"^": {
+			precedence: 4,
+			associativity: "Right",
+		},
+		"/": {
+			precedence: 3,
+			associativity: "Left",
+		},
+		"*": {
+			precedence: 3,
+			associativity: "Left",
+		},
+		"+": {
+			precedence: 2,
+			associativity: "Left",
+		},
+		"-": {
+			precedence: 2,
+			associativity: "Left",
+		},
+	};
+	infix = infix.replace(/\s+/g, "");
+	infix = infix.split(/([+\-*/^()])/);
+	infix = infix.filter(token => token);
+	let i;
+	for (i = 0; i < infix.length; i++) {
+		let token = infix[i];
+		if (isNumeric(token) === true) {
+			outputQueue.push(token);
+		} else if ("^*/+-".indexOf(token) !== -1) {
```

Comment:
```suggestion
		} else if ("^*/+-".includes(token)) {
```

PS supports ES6, which is more readable here.